packages/server/lib/util/specs.ts

```diff
@@ -161,90 +163,85 @@ function findSpecsOfType (searchOptions, specPattern) {
-
-  const findIntegrationSpecs = () => {
-    const searchOptions = _.pick(config, commonSearchOptions)
-
-    // ? should we always use config.resolved instead of config?
-    searchOptions.searchFolder = config.integrationFolder
+  return findSpecsOfType(searchFolder, commonSearchOptions, specPattern)
+  .then(setTestType(SPEC_TYPES.INTEGRATION))
+}
 
-    return findSpecsOfType(searchOptions, specPattern)
-    .then(setTestType(SPEC_TYPES.INTEGRATION))
+const findComponentSpecs = (searchFolder: string | undefined | false, commonSearchOptions: CommonSearchOptions, specPattern: string | undefined) => {
+  if (!searchFolder) {
+    return []
   }
 
-  const findComponentSpecs = () => {
-    // ? should we always use config.resolved instead of config?
-    if (!config.componentFolder) {
-      return []
-    }
+  return findSpecsOfType(searchFolder, commonSearchOptions, specPattern)
+  .then(setTestType(SPEC_TYPES.COMPONENT))
+}
 
-    const searchOptions = _.pick(config, commonSearchOptions)
+const printFoundSpecs = (foundSpecs: Cypress.Spec[]) => {
+  const table = new Table({
+    head: ['relative', 'specType'],
+  })
 
-    searchOptions.searchFolder = config.componentFolder
+  foundSpecs.forEach((spec: Cypress.Spec) => {
+    // @ts-ignore - types are incorrect
+    table.push([spec.relative, spec.specType])
+  })
 
-    return findSpecsOfType(searchOptions, specPattern)
-    .then(setTestType(SPEC_TYPES.COMPONENT))
-  }
+  /* eslint-disable no-console */
+  console.error(table.toString())
+}
 
-  const printFoundSpecs = (foundSpecs) => {
-    const table = new Table({
-      head: ['relative', 'specType'],
-    })
+export interface CommonSearchOptions {
+  projectRoot: Cypress.RuntimeConfigOptions['projectRoot']
+  fixturesFolder: Cypress.ResolvedConfigOptions['fixturesFolder']
+  supportFile: Cypress.ResolvedConfigOptions['supportFile']
+  testFiles: Cypress.ResolvedConfigOptions['testFiles']
+  ignoreTestFiles: Cypress.ResolvedConfigOptions['ignoreTestFiles']
+}
 
-    foundSpecs.forEach((spec) => {
-      table.push([spec.relative, spec.specType])
-    })
+type FindSpecs = {
+  componentFolder: Cypress.ResolvedConfigOptions['componentFolder']
+  integrationFolder: Cypress.ResolvedConfigOptions['integrationFolder']
+} & CommonSearchOptions
 
-    /* eslint-disable no-console */
-    console.error(table.toString())
-  }
+/**
+ * First, finds all integration specs, then finds all component specs.
+ * Resolves with an array of objects. Each object has a "testType" property
+ * with one of TEST_TYPES values.
+ */
+const findSpecs = (payload: FindSpecs, specPattern?: string) => {
+  const { componentFolder, integrationFolder, ...commonSearchOptions } = payload
+
+  return Bluebird.all([
+    findComponentSpecs(componentFolder, commonSearchOptions, specPattern),
+    findIntegrationSpecs(integrationFolder, commonSearchOptions, specPattern),
+  ]).then(([ct, e2e]) => {
```

Comment:
This change is the important and interesting part, and the main motivation behind this PR. I'd like to use `findSpecs` in the work I'm doing in unified GUI, and we would like to get a list of *all* the specs. At the moment, `findSpecs` either returns CT or E2E - I'd like it to be less opinionated, returning all the specs, then filtering afterwards if/when needed. 

In the future, it might be nice support a third argument to `findSpecs` to filter by a spec type, but it's not really necessary right now.